src/app/cli/src/cli_entrypoint/mina_cli_entrypoint.ml

```diff
@@ -1918,6 +1918,117 @@ let internal_commands logger =
   ; ("dump-type-shapes", dump_type_shapes)
   ; ("replay-blocks", replay_blocks logger)
   ; ("audit-type-shapes", audit_type_shapes)
+  ; ( "test-genesis-block-generation"
+    , Command.async ~summary:"Generate a genesis proof"
+        (let open Command.Let_syntax in
+        let%map_open config_file =
+          flag "--config-file" ~aliases:[ "config-file" ]
+            ~doc:
+              "PATH path to a configuration file (overrides MINA_CONFIG_FILE, \
+               default: <config_dir>/daemon.json). Pass multiple times to \
+               override fields from earlier config files"
+            (required string)
+        and conf_dir = Cli_lib.Flag.conf_dir
+        and genesis_dir =
+          flag "--genesis-ledger-dir" ~aliases:[ "genesis-ledger-dir" ]
+            ~doc:
+              "DIR Directory that contains the genesis ledger and the genesis \
+               blockchain proof (default: <config-dir>)"
+            (optional string)
+        in
+        fun () ->
+          let open Deferred.Let_syntax in
+          Parallel.init_master () ;
+          let logger = Logger.create () in
+          let conf_dir = Mina_lib.Conf_dir.compute_conf_dir conf_dir in
+          let genesis_dir =
+            Option.value ~default:(conf_dir ^/ "genesis") genesis_dir
+          in
+          let proof_level = Genesis_constants.Proof_level.Full in
+          let%bind config =
```

Comment:
Would be nice if we could share some code here with other commands that accept a config file arg. Specifically, if this code was shared with the daemon command, we would have have more confidence that this command succeeding guarantees the daemon will be able to produce the genesis proof as well, since we know the routine for initializing the configuration is the same across the 2 commands.